### PR TITLE
boards: arm: stm32: Update support for STM32F746G Discovery board

### PR DESCRIPTION
**Pull request by gromero**
Hi,

Could the following change be reviewed please?

Support for STM32F746G Discovery board was updated on 'openocd' by commit [0]:

```
  commit f79c902686f3ae500f3426dc2e552fdec80796d1
  Author: Andreas Bolsch <hyphen0break@gmail.com>
  Date:   Wed Dec 21 10:35:58 2016 +0100

      Flash, FRAM and EEPROM driver for STM32 QUAD-/OCTOSPI interface
```

which among other comments reads:
```
    - tcl/board/stm32f7discovery.cfg removed as name is ambiguous
      (superseded by stm32f746g-disco.cfg vs. stm32f769i-disco.cfg)
```

Thus that commit replaces the use of stm32f7discovery.cfg (superseded) by
stm32f746g-disco.cfg on Zephyr so the updated config file is used instead
of the current superseded one that sets incorrectly (lower) the adapter
speed when flashing the board (2000 kHz vs 4000 kHz).

One of the goals of that change is get rid of messages like the following when flashing:

```
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
Info : Unable to match requested speed 2000 kHz, using 1800 kHz
```

That change is also related to openocd PR 41 [1].

Thanks,
Gustavo

[0] https://github.com/zephyrproject-rtos/openocd/commit/f79c902686f3ae500f3426dc2e552fdec80796d1
[1] https://github.com/zephyrproject-rtos/openocd/pull/41

### CODE REVIEW

**Comment by microbuilder** [comment]
This works with the Zephyr SDK, which has `board/stm32f746g-disco.cfg`, but the file may not be available on platforms where the SDK is not being used (Windows, OS X).

Using the latest 0.10.0 `brew` version of `openocd`, for example, we have the previous `stm32f7discovery.cfg` file but not the board-specific one.

This may not be a valid argument NOT to make this change, but finding a way to include the new config file in the board folder for Zephyr may be a better option that maintains out-of-the-box support on all platforms???

```bash
$ west flash
-- west flash: rebuilding
[0/1] cd [PATH]/zephyr/upstream/build/zeph...ash && /usr/local/Cellar/cmake/3.18.1/bin/cmake -E ech

-- west flash: using runner openocd
-- runners.openocd: Flashing file: [PATH]/zephyr/upstream/build/zephyr/zephyr.hex
Open On-Chip Debugger 0.10.0
Licensed under GNU GPL v2
For bug reports, read
        http://openocd.org/doc/doxygen/bugs.html
[PATH]/zephyr/boards/arm/stm32f746g_disco/support/openocd.cfg:1: Error: Can't find board/stm32f746g-disco.cfg
in procedure 'script' 
at file "embedded:startup.tcl", line 60
at file "[PATH]/zephyr/boards/arm/stm32f746g_disco/support/openocd.cfg", line 1
FATAL ERROR: command exited with status 1: /usr/local/bin/openocd -s [PATH]/zephyr/boards/arm/stm32f746g_disco/support -f [PATH]/zephyr/boards/arm/stm32f746g_disco/support/openocd.cfg -c init -c targets -c 'reset halt' -c 'flash write_image erase [PATH]zephyr/zephyr.hex' -c 'reset halt' -c 'verify_image [PATH]/zephyr/zephyr.hex' -c 'reset run' -c shutdown
```

The available files for brew's 0.10.0 release of openocd are here:

```bash
$ ls /usr/local/Cellar/open-ocd/0.10.0/share/openocd/scripts/board/
actux3.cfg                           hilscher_nxhx10.cfg                  spear300evb_mod.cfg
adapteva_parallella1.cfg             hilscher_nxhx50.cfg                  spear310evb20.cfg
alphascale_asm9260_ek.cfg            hilscher_nxhx500.cfg                 spear310evb20_mod.cfg
altera_sockit.cfg                    hilscher_nxsb100.cfg                 spear320cpu.cfg
am3517evm.cfg                        hitex_lpc1768stick.cfg               spear320cpu_mod.cfg
arm_evaluator7t.cfg                  hitex_lpc2929.cfg                    st_nucleo_f0.cfg
asus-rt-n16.cfg                      hitex_stm32-performancestick.cfg     st_nucleo_f103rb.cfg
asus-rt-n66u.cfg                     hitex_str9-comstick.cfg              st_nucleo_f3.cfg
at91cap7a-stk-sdram.cfg              iar_lpc1768.cfg                      st_nucleo_f4.cfg
at91eb40a.cfg                        iar_str912_sk.cfg                    st_nucleo_l1.cfg
at91rm9200-dk.cfg                    icnova_imx53_sodimm.cfg              st_nucleo_l476rg.cfg
at91rm9200-ek.cfg                    icnova_sam9g45_sodimm.cfg            steval_pcc010.cfg
at91sam9261-ek.cfg                   imx27ads.cfg                         stm320518_eval.cfg
at91sam9263-ek.cfg                   imx27lnst.cfg                        stm320518_eval_stlink.cfg
at91sam9g20-ek.cfg                   imx28evk.cfg                         stm32100b_eval.cfg
atmel_at91sam7s-ek.cfg               imx31pdk.cfg                         stm3210b_eval.cfg
atmel_at91sam9260-ek.cfg             imx35pdk.cfg                         stm3210c_eval.cfg
atmel_at91sam9rl-ek.cfg              imx53-m53evk.cfg                     stm3210e_eval.cfg
atmel_sam3n_ek.cfg                   imx53loco.cfg                        stm3220g_eval.cfg
atmel_sam3s_ek.cfg                   insignal_arndale.cfg                 stm3220g_eval_stlink.cfg
atmel_sam3u_ek.cfg                   kc705.cfg                            stm3241g_eval.cfg
atmel_sam3x_ek.cfg                   keil_mcb1700.cfg                     stm3241g_eval_stlink.cfg
atmel_sam4e_ek.cfg                   keil_mcb2140.cfg                     stm32429i_eval.cfg
atmel_sam4l8_xplained_pro.cfg        kwikstik.cfg                         stm32429i_eval_stlink.cfg
atmel_sam4s_ek.cfg                   la_fonera-fon2200.cfg                stm32439i_eval.cfg
atmel_sam4s_xplained_pro.cfg         linksys-wag200g.cfg                  stm32439i_eval_stlink.cfg
atmel_samc20_xplained_pro.cfg        linksys-wrt54gl.cfg                  stm327x6g_eval.cfg
atmel_samc21_xplained_pro.cfg        linksys_nslu2.cfg                    stm32f0discovery.cfg
atmel_samd20_xplained_pro.cfg        lisa-l.cfg                           stm32f334discovery.cfg
atmel_samd21_xplained_pro.cfg        logicpd_imx27.cfg                    stm32f3discovery.cfg
atmel_same70_xplained.cfg            lpc1850_spifi_generic.cfg            stm32f429disc1.cfg
atmel_samg53_xplained_pro.cfg        lpc4350_spifi_generic.cfg            stm32f429discovery.cfg
atmel_saml21_xplained_pro.cfg        lubbock.cfg                          stm32f469discovery.cfg
atmel_samr21_xplained_pro.cfg        marsohod.cfg                         stm32f4discovery.cfg
atmel_samv71_xplained_ultra.cfg      marsohod2.cfg                        stm32f7discovery.cfg
balloon3-cpu.cfg                     marsohod3.cfg                        stm32l0discovery.cfg
bcm28155_ap.cfg                      mbed-lpc11u24.cfg                    stm32l4discovery.cfg
bt-homehubv1.cfg                     mbed-lpc1768.cfg                     stm32ldiscovery.cfg
colibri.cfg                          mcb1700.cfg                          stm32vldiscovery.cfg
crossbow_tech_imote2.cfg             microchip_explorer16.cfg             str910-eval.cfg
csb337.cfg                           mini2440.cfg                         telo.cfg
csb732.cfg                           mini6410.cfg                         ti-cc3200-launchxl.cfg
da850evm.cfg                         minispartan6.cfg                     ti_am335xevm.cfg
digi_connectcore_wi-9c.cfg           nds32_xc5.cfg                        ti_am437x_idk.cfg
digilent_analog_discovery.cfg        netgear-dg834v3.cfg                  ti_am43xx_evm.cfg
digilent_atlys.cfg                   netgear-wg102.cfg                    ti_beagleboard.cfg
digilent_zedboard.cfg                nordic_nrf51822_mkit.cfg             ti_beagleboard_xm.cfg
diolan_lpc4350-db1.cfg               nordic_nrf51_dk.cfg                  ti_beaglebone.cfg
diolan_lpc4357-db1.cfg               nordic_nrf52_dk.cfg                  ti_blaze.cfg
dk-tm4c129.cfg                       novena-internal-fpga.cfg             ti_pandaboard.cfg
dm355evm.cfg                         numato_opsis.cfg                     ti_pandaboard_es.cfg
dm365evm.cfg                         nxp_lpc-link2.cfg                    ti_tmdx570ls20susb.cfg
dm6446evm.cfg                        olimex_LPC2378STK.cfg                ti_tmdx570ls31usb.cfg
dp_busblaster_v3.cfg                 olimex_lpc_h2148.cfg                 topas910.cfg
efikamx.cfg                          olimex_sam7_ex256.cfg                topasa900.cfg
efm32.cfg                            olimex_sam7_la2.cfg                  tp-link_tl-mr3020.cfg
eir.cfg                              olimex_sam9_l9260.cfg                twr-k60f120m.cfg
ek-lm3s1968.cfg                      olimex_stm32_h103.cfg                twr-k60n512.cfg
ek-lm3s3748.cfg                      olimex_stm32_h107.cfg                twr-vf65gs10.cfg
ek-lm3s6965.cfg                      olimex_stm32_p107.cfg                twr-vf65gs10_cmsisdap.cfg
ek-lm3s811-revb.cfg                  omap2420_h4.cfg                      tx25_stk5.cfg
ek-lm3s811.cfg                       open-bldc.cfg                        tx27_stk5.cfg
ek-lm3s8962.cfg                      openrd.cfg                           unknown_at91sam9260.cfg
ek-lm3s9b9x.cfg                      or1k_generic.cfg                     uptech_2410.cfg
ek-lm3s9d92.cfg                      osk5912.cfg                          verdex.cfg
ek-lm4f120xl.cfg                     phone_se_j100i.cfg                   voipac.cfg
ek-lm4f232.cfg                       phytec_lpc3250.cfg                   voltcraft_dso-3062c.cfg
ek-tm4c123gxl.cfg                    pic-p32mx.cfg                        x300t.cfg
ek-tm4c1294xl.cfg                    pipistrello.cfg                      xmc-2go.cfg
embedded-artists_lpc2478-32.cfg      propox_mmnet1001.cfg                 xmc1100-boot-kit.cfg
emcraft_twr-vf6-som-bsb.cfg          pxa255_sst.cfg                       xmc4200-application-kit-actuator.cfg
emcraft_vf6-som.cfg                  quark_d2000_refboard.cfg             xmc4300-relax.cfg
ethernut3.cfg                        quark_x10xx_board.cfg                xmc4500-application-kit-general.cfg
frdm-kl25z.cfg                       redbee.cfg                           xmc4500-application-kit-sdram.cfg
frdm-kl46z.cfg                       renesas_dk-s7g2.cfg                  xmc4500-relax.cfg
fsl_imx6q_sabresd.cfg                rsc-w910.cfg                         xmc4700-relax.cfg
glyn_tonga2.cfg                      sheevaplug.cfg                       xmc4800-relax.cfg
gumstix-aerocore.cfg                 smdk6410.cfg                         xmos_xk-xac-xa8_arm.cfg
hammer.cfg                           spansion_sk-fm4-176l-s6e2cc.cfg      zy1000.cfg
hilscher_nxdb500sys.cfg              spansion_sk-fm4-u120-9b560.cfg
hilscher_nxeb500hmi.cfg              spear300evb.cfg
```

**Comment by gromero** [comment]
@microbuilder Thanks a lot for checking it on other environments! What you find actually weights in to make the chance I'm proposing. The error you found even happens on Linux and using `zephyr-sdk-0.11.4` (the one we've been using), because the commit I pointed out `f79c902686f3ae500f3426dc2e552fdec80796d1` is not present. It was fixed (added) later, so that change should be applied at least to `zephyr-sdk-0.12.1`, where the commit is applied and so file `stm32f746g-disco.cfg` is preset. Hence, the issue you found, in my understanding, is actually related to Zephyr SDK version than to the platform used. Would it be possible to make the same test you did above  but using `zephyr-sdk-0.12.1` SDK instead?

**Comment by microbuilder** [comment]
> Would it be possible to make the same test you did above but using `zephyr-sdk-0.12.1` SDK instead?

The SDK is normally only used on Linux, whereas a lot of people use native tools on OS X and Windows where this will remain a problem. I can also test this with the SDK, though.

Thanks for pointing out the specific commit, though. Looking at the [commit message](https://github.com/zephyrproject-rtos/openocd/commit/f79c902686f3ae500f3426dc2e552fdec80796d1) I can see:

```
- tcl/board/stm32f7discovery.cfg removed as name is ambiguous
  (superseded by stm32f746g-disco.cfg vs. stm32f769i-disco.cfg)
```

**Comment by erwango** [comment]
@gromero, diffing `stm32f746g-disco.cfg` and `stm32f7discovery.cfg`, the main diff found is about the QSPI init clock setup.
This has no impact on zephyr flashing as we're not using this (yet).

The flash speed adaptation that your proposed for openocd upstream in https://github.com/zephyrproject-rtos/openocd/pull/41, has no dependency on this file, so couldn't it be applied to `boards/arm/stm32f746g_disco/support/openocd.cfg` directly to remove this dependency on openocd ?

**Comment by microbuilder** [comment]
@galak Any thoughts on this? How have different openocd config files across releases/platforms been handled in the past?

**Comment by gromero** [comment]
> @gromero, diffing `stm32f746g-disco.cfg` and `stm32f7discovery.cfg`, the main diff found is about the QSPI init clock setup.
> This has no impact on zephyr flashing as we're not using this (yet).
> 
> The flash speed adaptation that your proposed for openocd upstream in [zephyrproject-rtos/openocd#41](https://github.com/zephyrproject-rtos/openocd/pull/41), has no dependency on this file, so couldn't it be applied to `boards/arm/stm32f746g_disco/support/openocd.cfg` directly to remove this dependency on openocd ?

I could but I think the correct way to do is to fix it on openocd upstream anyway, as pointed out by @galak (https://github.com/zephyrproject-rtos/openocd/pull/41).

Another detail is that `stm32f746g-disco.cfg` shouldn't even exist anymore since it was superseded by `stm32f746g-disco.cfg`. In my understanding commit `f79c902686f3ae500f3426dc2e552fdec80796d1` just missed that and so that file wrongly "propagated" downstream into Zephyr's OpenOCD, even after @galak applied  that commit to `zephyr-20200722` at the end of the last year (appearing now on `zephyr-sdk-0.12.1`.

**Comment by erwango** [comment]
@gromero Up to you. With a little experience on openocd upstream and backward intregration in various Zephyr environments (SDK for Linux, Binary availability on Windows and mac stuff), this can take few months before we can merge current change.
In the other hand doing the speed adaptation fix in zephyr openocd.cfg would work right now on all environments and could be kept until openocd upstream changes could be used (at this point, zephyr change could be reverted).
So actually, I don't say we should not do the upstream change in openocd. I just say we can do both.

**Comment by microbuilder** [comment]
It's your call @gromero, but I think I agree with @erwango that we can fix this locally now, and it will eventually trickle down from upstream, which is a much slower process.

**Comment by gromero** [comment]
@erwango oh got it! Let's do both so, that's better. I kind thought that same thing when submitting the change to Zephyr's openocd  because I didn't know when/how it would sync up with openocd upstream, like maybe Zephyr's openocd could take my change in whilst I would figure out how openocd upstream works. But the way you're is better, i.e. just apply a single change to Zephyr's local config file. I'll keep track of openocd upstream changes. Let me prepare a new PR to Zephyr. Thanks a lot for the guidance. Cheers.

**Comment by gromero** [comment]
> It's your call @gromero, but I think I agree with @erwango that we can fix this locally now, and it will eventually trickle down from upstream, which is a much slower process.

@microbuilder Right! : )

**Review by FRASTM — APPROVED** [review]

**Comment by erwango** [comment]
@gromero Is that ok to close this since https://github.com/zephyrproject-rtos/zephyr/pull/31430 has been merged ?

**Comment by gromero** [comment]
> @gromero Is that ok to close this since #31430 has been merged ?

Hi @erwango! That's what I was wondering right now : ) So, I think we can close it by now and wait for me to sort things out on openocd upstream. Then when `stm32f746g-disco.cfg` in openocd upstream gets the speed updated as in #31430 we could reopen it. What do you think?

**Comment by erwango** [comment]
@gromero fine with me !

**Pull request closed without merging**
